src/app/files/fileManager.js

```diff
@@ -42,6 +42,212 @@ class FileManager extends Plugin {
+   * @returns {boolean} true if the path exists
+   */
+  exists (path) {
+    const provider = this.fileProviderOf(path)
+    return provider ? true : false // eslint-disable-line
+  }
+
+  /**
+   * Verify if the path provided is a file
+   * @param {string} path path of the directory or file
+   * @returns {boolean} true if path is a file.
+   */
+  isFile (path) {
+    const extension = path.split('/').pop()
+
+    return extension && extension.indexOf('.') > -1
+  }
+
+  /**
+   * Open the content of the file in the context (eg: Editor)
+   * @param {string} path path of the file
+   * @returns {void}
+   */
+  open (path) {
+    this._handleExists(path, `Cannot open file ${path}`)
+    this._handleIsFile(path, `Cannot open file ${path}`)
+    return this.switchFile(path)
+  }
+
+  /**
+   * Set the content of a specific file
+   * @param {string} path path of the file
+   * @param {string} data content to write on the file
+   * @returns {void}
+   */
+  writeFile (path, data) {
+    this._handleIsFile(path, `Cannot write file ${path}`)
+    this.setFile(path, data)
+  }
+
+  /**
+   * Return the content of a specific file
+   * @param {string} path path of the file
+   * @returns {string} content of the file
+   */
+  readFile (path) {
+    this._handleExists(path, `Cannot read file ${path}`)
+    this._handleIsFile(path, `Cannot read file ${path}`)
+    return this.getFile(path)
+  }
+
+  /**
+   * Upsert a file with the content of the source file
+   * @param {string} src path of the source file
+   * @param {string} dest path of the destrination file
+   * @returns {void}
+   */
+  copyFile (src, dest) {
+    this._handleExists(src, `Cannot copy from ${src}`)
+    this._handleIsFile(src, `Cannot copy from ${src}`)
+    this._handleIsFile(dest, `Cannot paste content into ${dest}`)
+    const content = this.readFile(src)
+    this.writeFile(dest, content)
+  }
+
+  /**
+   * Removes a file
+   * @param {string} path path of the file to remove
+   * @note will not work on a directory, use `rmdir` instead
+   * @returns {void}
+   */
+  unlink (path) {
+    this._handleExists(path, `Cannot remove file ${path}`)
+    this._handleIsDir(path, `Cannot remove file ${path}`)
+  }
+
+  /**
+   * Change the path of a file/directory
+   * @param {string} oldPath current path of the file/directory
+   * @param {string} newPath new path of the file/directory
+   * @returns {void}
+   */
+  rename (oldPath, newPath) {
+    this.__handleExists(oldPath, `Cannot rename ${oldPath}`)
+    // todo: should we verify if newPath exists here ?
+    const isFile = this.isFile(oldPath)
+    this.fileRenamedEvent(oldPath, newPath, !isFile)
+  }
+
+  /**
+   * Create a directory
+   * @param {string} path path of the new directory
+   * @returns {void}
+   */
+  mkdir (path) {
+    if (this.exists(path)) {
+      this._handleError({ code: 'EEXIST', message: `Cannot create directory ${path}` })
+    }
+    // To implement
```

Comment:
that should do the trick `window.remixFileSystem.mkdirSync`
or look at https://github.com/ethereum/remix-ide/blob/master/src/app/files/fileProvider.js#L92 for recursively creating folder.